x/proof/keeper/msg_server_create_claim.go

```diff
@@ -16,20 +18,25 @@ func (k msgServer) CreateClaim(
 	msg *types.MsgCreateClaim,
 ) (_ *types.MsgCreateClaimResponse, err error) {
 	// Declare claim to reference in telemetry.
-	var claim types.Claim
+	var (
+		claim           types.Claim
+		isExistingClaim bool
+		numRelays       uint64
+		numComputeUnits uint64
+	)
 
 	// Defer telemetry calls so that they reference the final values the relevant variables.
 	defer func() {
-		// TODO_IMPROVE: We could track on-chain relays here with claim.GetNumRelays().
-		numComputeUnits, deferredErr := claim.GetNumComputeUnits()
-		err = errors.Join(err, deferredErr)
-
-		telemetry.ClaimCounter(telemetry.ClaimProofStageClaimed, 1, err)
-		telemetry.ClaimComputeUnitsCounter(
-			telemetry.ClaimProofStageClaimed,
-			numComputeUnits,
-			err,
-		)
+		// Only increment these metrics counters if handling a new claim.
+		if !isExistingClaim {
+			// TODO_IMPROVE: We could track on-chain relays here with claim.GetNumRelays().
+			telemetry.ClaimCounter(types.ClaimProofStage_CLAIMED, 1, err)
+			telemetry.ClaimComputeUnitsCounter(
+				types.ClaimProofStage_CLAIMED,
+				numComputeUnits,
+				err,
```

Comment:
Shouldn't we meter `numRelays` too?

Reply:
This is a telemetry call, not an event emission. This particular telemetry call is specific to compute units.

Reply:
I'm realizing that I misunderstood your message initially but I get it now. I'm not sure what the state of the code was at the time but now it looks like this:

```go
			telemetry.ClaimCounter(types.ClaimProofStage_CLAIMED, 1, err)
			telemetry.ClaimComputeUnitsCounter(
				types.ClaimProofStage_CLAIMED,
				numComputeUnits,
				err,
			)
```

Reply:
Nope. Did it again. 😅🙈

I've opened #643 to track this.